terraform/examples/io500/configure_daos.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+#
+# Configures the /etc/daos/daos_*.yml files on daos-server and daos-client
+# instances.
+#
+# TODO: Move everything in this script to Terraform and/or startup scripts.
+#
+
+SCRIPT_NAME="$(basename "$0")"
+SCRIPT_DIR="$( cd "$( dirname "$0" )" && pwd )"
+SCRIPT_COMPLETED_FILE="${SCRIPT_DIR}/${SCRIPT_NAME}.completed"
+CONFIG_FILE="${SCRIPT_DIR}/config.sh"
+
+# Source config file to load variables
+source "${CONFIG_FILE}"
+
+log() {
+  if [[ -t 1 ]]; then tput setaf 14; fi
+  printf -- "\n%s\n\n" "${1}"
+  if [[ -t 1 ]]; then tput sgr0; fi
+}
+
+DAOS_FIRST_SERVER=$(head -n 1 ~/hosts_servers)
+
+check_already_run() {
+  # Check to see if this script has already run
+  if [[ -f "${SCRIPT_COMPLETED_FILE}" ]]; then
+    # This script has already been run and doesn't need to run again
+    exit 0
+  fi
+}
+
+update_ssh_dir() {
+  # Clear ~/.ssh/known_hosts so we don't run into any issues
+  clush --hostfile=hosts_all --dsh 'rm -f ~/.ssh/known_hosts'
+
+  # Copy ~/.ssh directory to all instances
+  pdcp -w^hosts_all -r ~/.ssh ~/
+}
+
+configure_servers() {
+
+  echo "Getting /etc/daos/daos_server.yml from ${DAOS_FIRST_SERVER}"
+
+  scp ${DAOS_FIRST_SERVER}:/etc/daos/daos_server.yml "${SCRIPT_DIR}/"
+
+  echo "Updating daos_server.yml"
+
+  # Set nr_hugepages value
+  # nr_hugepages = (targets * 1Gib) / hugepagesize
+  #    Example: for 8 targets and Hugepagesize = 2048 kB:
+  #       Targets = 8
+  #       1Gib = 1048576 KiB
+  #       Hugepagesize = 2048kB
+  #       nr_hugepages=(8*1048576) / 2048
+  #       So nr_hugepages value is 4096
+  hugepagesize=$(ssh ${DAOS_FIRST_SERVER} "grep Hugepagesize /proc/meminfo | awk '{print \$2}'")
+  nr_hugepages=$(( (${DAOS_SERVER_DISK_COUNT}*1048576) / ${hugepagesize} ))
+  sed -i "s/^nr_hugepages:.*/nr_hugepages: ${nr_hugepages}/g" "${SCRIPT_DIR}/daos_server.yml"
+  sed -i "s/^crt_timeout:.*/crt_timeout: ${DAOS_SERVER_CRT_TIMEOUT}/g" "${SCRIPT_DIR}/daos_server.yml"
+  sed -i "s/^\(\s*\)targets:.*/\1targets: ${DAOS_SERVER_DISK_COUNT}/g" "${SCRIPT_DIR}/daos_server.yml"
+  sed -i "s/^\(\s*\)scm_size:.*/\1scm_size: ${DAOS_SERVER_SCM_SIZE}/g" "${SCRIPT_DIR}/daos_server.yml"
+
+  # Copy daos_server.yml to all servers
```

Comment:
@markaolson  This type of things needs to be apart of the Server code not of any IO500 Example.  Please refactor this logic into Server cod

Reply:
I sent you an e-mail about this. You may have not seen it yet.  The TODO comment at the top of the file was added because, yes, everything in this file this needs to move to terraform/startup scripts.  However, it's too big of a change to include in this PR. Although this is a new file this code already existed in different files. I didn't add this feature. I just moved it to this file and called clush to deploy the .yml files instead of looping in a for loop and calling SSH for each server. I looked for code that manipulated daos_*.yml files in different scripts in the terraform/examples/io500 directory and then consolidated it into this single file. That way when we actually do move the configuration to terraform/startup scripts it should be easy to yank this out by deleting this one file and then removing the line in start.sh that calls it.  I think the refactoring you are requesting should be done in another PR to focus specifically on those changes rather than make this PR larger.  @cboneti and @johannlombardi do you agree?

Reply:
@markaolson  I agree this patch is huge.  Breaking things into smaller pieces. helps. 

Can we open an issue to track IO500 work somewhere like we have been talking about for few weeks?   If so then I can agree to a fix in future version type way.